src/main/java/net/sf/jabref/logic/util/io/FileUtil.java

```diff
@@ -407,7 +407,7 @@ public static String createFileNameFromPattern(BibDatabase database, BibEntry en
         }
         //Removes illegal characters from filename
         targetName = FileNameCleaner.cleanFileName(targetName);
-        return targetName;
+        return targetName.trim();
```

Comment:
Good spot! I oversaw it. Btw, the method cleanFileName was the root cause for the problem